include/sparseLDL/ldl.h

```diff
@@ -0,0 +1,173 @@
+#pragma once
+
+#include <stddef.h>
+
+#include "sparseLDL/ldl_types.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+struct qp_size {
+  int *nx;  // number of states
+  int *nu;  // number of inputs
+  int *ng;  // number of constraints
+  int N;    // horizon length
+  size_t memsize;
+};
+
+struct qp_data {
+  // Dynamics
+  ldl_matrix *A;
+  ldl_matrix *B;
+  ldl_matrix *C;
+  ldl_matrix *D;
+
+  // Cost
+  ldl_matrix *Q;
+  ldl_matrix *R;
+  ldl_matrix *H;
+  ldl_vector *q;
+  ldl_vector *r;
+
+  // Constraints
+  ldl_matrix *E;
+  ldl_matrix *F;
+  ldl_vector *g;
+
+  size_t memsize;
+};
+
+struct ldl_workspace {
+  ldl_matrix *L;
```

Comment:
I don't think the struct definitions should be part of the API. Use a forward-declaration instead to hide the implementation details from the user.

Reply:
If we are going to expose 

```
struct ldl_Solver {
  qp_size* size;
  qp_data* data;
  ldl_workspace ws;
};
```
to users, the definition of member variables should also be visible to users, right?